### PR TITLE
Merge from toCode to master

### PR DESCRIPTION
**Pull request by tgargula**

### CODE REVIEW

File: ASD/graph/DFS.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    for u in V:
+        if u.color == "white":
+            time = DFSvisit(G, u, time)
+        
+def DFSvisit(G, u, time):
+    time += 1
+    u.t1 = time
+    u.color = "grey"
+    for v in G.neighbors(u):
+        if v.color == "white":
```
Comment (jakub-gonet) [inline ASD/graph/DFS.py:30]: Why do you need those colors? `is_visited` field should suffice

File: ASD/graph/topological_sorting.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            DFS(G, L, u)
+    L.reverse()
+    return L
+
+def DFS(G, L, u):
+    u.color = "grey"
+    for v in G.neighbors(u):
+        if v.color == "grey":   # CYCLE
+            print("Graph is not a DAG")
+            sys.exit()
```
Comment (jakub-gonet) [inline ASD/graph/topological_sorting.py:31]: you should raise an exception here

File: ASD/graph/DFS.py (hunk 1 above)
Comment (tgargula) [reply ASD/graph/DFS.py:30]: I've written this file as DFS template so I decided not to replace 'colors' with 'visited'

**Review by Wolokin — APPROVED** [review]
Cosmetics ;)

**Review by jgonet — APPROVED** [review]

File: PI/Miller/10/linkedListFromTxt.c [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+            word[i] += 32;
+        }
+        if (isValidCharacter(word[i])) {
+            word[i2++] = word[i];
+        }
+    }
+    
+    word[i2] = '\0';
+    if (isLetter(word[0]) == 0) return; 
+    char *s = (char*)malloc(strlen(word) * sizeof(char));
```
Comment (tgargula) [inline PI/Miller/10/linkedListFromTxt.c:70]: It is unnecessary if you use char s[MAXCHAR].
But if you want to allocate it dynamically:
`char \*s = (char\*)malloc(strlen(word) + 1) * sizeof(char));`
This allocated memory is not freed!

File: PI/Miller/10/linkedListFromTxt.c [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    
+    node* listIterator = first;
+    node* previous = NULL;
+
+    while(listIterator != NULL && strcmp(word, listIterator->word) > 0) {
+        previous = listIterator; listIterator = listIterator->next;
+    }
+    
+    if(listIterator != NULL && strcmp(word, listIterator->word) == 0) {
+        listIterator->counter += 1; return first;
```
Comment (tgargula) [inline PI/Miller/10/linkedListFromTxt.c:31]: Same as above

File: PI/Miller/10/linkedListFromTxt.c [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+        if(first == NULL) {
+            printf("Word does not found in list\n"); return;
+        }
+    }
+
+    printf("Node order number: %d\n", listIterator);
+    printf("Number of word occurences: %d\n", first->counter);
+}
+
+int main() {
```
Comment (tgargula) [inline PI/Miller/10/linkedListFromTxt.c:132]: It is better to write:
`int main(void) {`

File: ASD/graph/BFS.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    Q.put(s)
+    while not Q.empty():
+        u = Q.get()
+        for v in G.neighbors(u):
+            if v.color == "white":
+                v.color = "grey"
+                v.d = u.d + 1
+                v.parent = u
+                Q.put(v)
+        u.color = "black"
```
[the remark this answers is not in the document]
Comment (tgargula) [reply ASD/graph/BFS.py:35]: But it actually is.
At first we put `s` in the empty queue. Then process all of its neighbours and at last we change its color to black.

File: ASD/graph/BFS.py [hunk 7]
```diff
@@ -0,0 +1,54 @@
+import queue
+
+inf = float('Inf')
+
+class Node:
+    def __init__(self, value):
+        self.value = value
+        self.parent = None
+        self.d = inf
+        self.color = "white"
```
[the remark this answers is not in the document]
Comment (tgargula) [reply ASD/graph/BFS.py:10]: I agree with that, but I have written this code as a BFS template

File: PI/Miller/10/linkedListFromTxt.c [hunk 8]
```diff
@@ -95,10 +99,11 @@ node* processFile(node* first) {                        // przetwarzanie pliku
     }
 
     fclose(txtfile);
+    free(inputFileName);
```
Comment (tgargula) [inline PI/Miller/10/linkedListFromTxt.c:102]: Invalid free – inputFileName was not dynamically allocated

**Pull request merged**